config/routes.rb

```diff
@@ -2,6 +2,9 @@
   crud = %i(index show create update destroy)
 
   namespace :v1 do
-    resources :projects, only: crud
+    resources :projects,  only: crud do
+      resources :tasks, only: [:index, :new, :create]
+    end
+    resources :tasks, only: [:show, :edit, :update, :destroy]
```

Comment:
An API typically doesn't have a :edit end point and you don't have one defined in your controller either, so it would be a good idea not to include it here.